packages/perseus/src/util/parse-perseus-json/general-purpose-parsers/object.ts

```diff
@@ -0,0 +1,24 @@
+import {isObject} from "./is-object";
+
+import type {ParsedValue, Parser} from "../parser-types";
+
+type ObjectSchema = Record<keyof any, Parser<any>>;
+
+export function object<S extends ObjectSchema>(
+    schema: S,
+): Parser<{[K in keyof S]: ParsedValue<S[K]>}> {
+    return (rawValue, ctx) => {
+        if (!isObject(rawValue)) {
+            return ctx.failure("object", rawValue);
+        }
+
+        for (const [prop, propParser] of Object.entries(schema)) {
+            const result = propParser(rawValue[prop], ctx.forSubtree(prop));
+            if (result.type === "failure") {
+                return result;
+            }
+        }
+
+        return ctx.success(rawValue as {[K in keyof S]: ParsedValue<S[K]>});
```

Comment:
Bug: in order for defaulting and other processing functions to work, we need to return the parsed values here instead of rawValue.

```ts
const ret: {[K in keyof S]: ParsedValue<S[K]>} = {};
for (const [prop, propParser] of Object.entries(schema)) {
  const result = propParser(rawValue[prop], ctx.forSubtree(prop));
  if (isFailure(result)) {
    return result;
  } else {
    ret[prop] = result.value;
  }
}

return ctx.success(ret);
```

Reply:
This is fixed now!